pkg/resiliency/policy_test.go

```diff
@@ -54,15 +54,14 @@ func TestPolicy(t *testing.T) {
 	ctx := context.Background()
 	for name, tt := range tests {
 		t.Run(name, func(t *testing.T) {
-			called := false
-			fn := func(ctx context.Context) error {
-				called = true
-
-				return nil
+			called := atomic.Bool{}
```

Comment:
Not really the best place for this comment but it's contextually relevant -- Can we simulate double-activation (or is that tested elsewhere?) to ensure that we handle race conditions now?